uber_sante/services/availability_service.py

```diff
@@ -9,63 +9,85 @@ def __init__(self):
         self.db = DBUtil.get_instance()
 
     def get_availabilities(self, schedule_request):
+        """
+        Queries the Availability table according to the schedule_request object
+        which specifies to query for the month or for a specific day only,
+        and for annual, walkin, or all booking types.
+        """
 
         date = schedule_request.get_request_date()
         year = date.get_year()
         month = date.get_month()
+        booking_type = schedule_request.get_appointment_request_type_value()
 
         if schedule_request.is_daily_request():
+
             day = date.get_day()
-            select_stmt = 'SELECT * FROM Availability WHERE year = ? AND month = ? AND day = ?'
-            params = (year, month, day)
+
+            # If the booking_type is BookingType.ALL (which has value ""), it will select all rows
+            select_stmt = 'SELECT * FROM Availability ' \
+                          'WHERE year = ? AND month = ? AND day = ?' \
+                          'AND (? = "ALL" OR booking_type = ?)'
+
+            params = (year, month, day, booking_type, booking_type)
 
         elif schedule_request.is_monthly_request():
-            select_stmt = 'SELECT * FROM Availability WHERE year = ? AND month = ?'
-            params = (year, month)
 
-        results = self.db.read_all(select_stmt, params)  # Returns a list of the matching rows
+            # If the booking_type is BookingType.ALL (which has value ""), it will select all rows
+            select_stmt = 'SELECT * FROM Availability ' \
+                          'WHERE year = ? AND month = ?' \
+                          'AND (? = "ALL" OR booking_type = ?)'
+
+            params = (year, month, booking_type, booking_type)
+
+        results = self.db.read_all(select_stmt, params)
 
         list_of_availabilities = []
+
         for result in results:
             list_of_availabilities.append(Availability(result['id'], result['doctor_id'], result['start'],
                                                        result['room'], result['free'], result['year'], result['month'],
                                                        result['day']))
 
         return list_of_availabilities
 
+    """
     def free_availabilities(self, availability_ids):
 
         for availability_id in availability_ids:
             update_stmt = 'UPDATE Availability SET free = 1 WHERE id = ?'
             params = (availability_id,)
             self.db.write_one(update_stmt, params)
+    """
+
+    def free_availability(self, availability_id):
+
+        update_stmt = 'UPDATE Availability SET free = 1 WHERE id = ?'
+        params = (availability_id, )
+        self.db.write_one(update_stmt, params)
 
     def validate_availability_and_reserve(self, appointment):
+        """
+        Given either a Walkin (20min, 1 row) or Annual appointment (60min, 3 rows),
```

Comment:
All appointments are now considered as 1 row with an attribute 'type' for annual or walkin, in the availability table that the doctors will fill with their availabilities

Reply:
You're right, I wasnt sure about this change since I missed the design. Ill change it

Reply:
Some diagrams are updated( the ones with last modified date 16 feb and later), the rest might not reflect these changes yet